service-base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/device/DeviceReadResult.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.service.base.jdbc.store.device;
+
+import java.util.Optional;
+
+import org.eclipse.hono.service.management.device.Device;
+
+import com.google.common.base.MoreObjects;
+
+public class DeviceReadResult {
```

Comment:
Could we use `OperationResult` class here?

Reply:
I would like to avoid that. First of all, that would introduce a dependency on the higher level stuff, to this lower level module. Also would the other fields (status, cache directive) have no meaning on this level.